### PR TITLE
[DA-2026] Migrations - Collecting Cloud Function Payloads

### PR DESCRIPTION
**Pull request by dtharpeuno**
## Resolves *[ticket DA-2026]*
https://precisionmedicineinitiative.atlassian.net/browse/DA-2026

## Description of changes/additions
-- Adds model/dao for storage in RDR for Genomic CF payloads
-- Edits cloud functions for new payload items and restructuring items in conditionals
-- Unit tests for the manifest route to collect payloads in db and missing API task routes 

## Tests
- [x] unit tests

### CODE REVIEW

File: rdr_service/dao/genomics_dao.py [hunk 1]
```diff
@@ -1623,3 +1623,19 @@ def truncate_value(value, max_length):
             value = value[:max_length]
 
         return is_truncated, value
+
+
+class GenomicCloudRequestsDao(UpdatableDao):
+
+    validate_version_match = False
```
Comment (pegbertsch) [inline rdr_service/dao/genomics_dao.py:1630]: I'm not clear on how/where this is used?  (although I see it defined in other classes as well)

Comment (dtharpeuno) [reply rdr_service/dao/genomics_dao.py:1630]: In the `UpdatableDao`:
https://github.com/all-of-us/raw-data-repository/blob/21baa86938f6a23527e079ff070b29eb1dc63b63/rdr_service/dao/base_dao.py#L668

**Review by pegbertsch — APPROVED** [review]

File: rdr_service/model/genomics.py [hunk 2]
```diff
@@ -629,3 +629,24 @@ class GenomicIncident(Base):
 
 event.listen(GenomicIncident, 'before_insert', model_insert_listener)
 event.listen(GenomicIncident, 'before_update', model_update_listener)
+
+
+class GenomicCloudRequests(Base):
+    """
+    """
```
Comment (wangy70) [inline rdr_service/model/genomics.py:638]: Do you want to add a comment for this class?

**Review by wangy70 — APPROVED** [review]

File: rdr_service/model/genomics.py (hunk 2 above)
Comment (dtharpeuno) [reply rdr_service/model/genomics.py:638]: Yes I do.

Comment (dtharpeuno) [reply rdr_service/model/genomics.py:638]: https://github.com/all-of-us/raw-data-repository/pull/2456/commits/2350c4383e93fba0bcec1843bfa8e33d4a8b7f26

File: rdr_service/api/genomic_cloud_tasks_api.py [hunk 3]
```diff
@@ -28,12 +28,19 @@ class BaseGenomicTaskApi(Resource):
...
     @task_auth_required
     def post(self):
         log_task_headers()
         self.data = request.get_json(force=True)
 
+    def create_cloud_record(self):
+        if self.data.get('cloud_function'):
+            insert_data = {key: value for key, value in self.data.items() if key in
+                           GenomicCloudRequests.__table__.columns.keys()}
+            self.cloud_req_dao.insert(GenomicCloudRequests(**insert_data))
```
Comment (j-kanuch) [inline rdr_service/api/genomic_cloud_tasks_api.py:42]: nice

**Review by j-kanuch — COMMENTED** [review]
Looks good!

**Pull request merged**